repos/jsutils/src/method/runSeq.js

```diff
@@ -0,0 +1,38 @@
+/** @module functions */
+
+const { validate } = require('../validation/validate')
+const { isArr } = require('../array/isArr')
+const { isFunc } = require('./isFunc')
+const { deepClone } = require('../collection/deepClone')
+
+/**
+* Calls each promise-returning function in array `asyncFns`,
+* but awaits each before calling the next. Will pass the
+* index and resolved values of complete functions to each subsequent
+* function, in case any need them.
+* @function
+* @param {Array<Function>} asyncFns - array of functions to call in sequence. 
+* @param {boolean?} cloneResults - (optional) if true, each function will be 
+* passed a deep clone of the results array, rather than the reference to it.
+* Each will be passed (currentIndex, resultsSoFar)
+* @return {Promise<Array<*>>} - returns a promise that resolves to an array of all the 
+* asyncFns' return values. If any element in asyncFns was not a function, that element's 
```

Comment:
Thanks, I agree about the configurable return value.  I went ahead and added that 👍🏼 .

I had considered the spread syntax for the args, but I prefer this because:
* I think this function will most often be called in cases in which you already have an array of functions to call. So it would be simpler to keep the first arg as an array
  * in a case of just a few functions, calling runSeq wouldn't make much sense. The user probably would just want to do `await` calls individually like:
```js
  await http.get()
  await updateApi()
  ...
```
* it's also more expensive. This could potentially be a large array of functions and we would need to create a new array from `args.slice`, + whatever the spread syntax might be doing)

Reply:
Yah, I see your point. 
When I said I was planning to write something like this, it was for ternary statement. For example =>

```js

  return error
    ? runSeq(
        () => console.log(error),
        () => false
      )[1]
    : runSeq(
        () => console.log(data),
        () => data
      )[1]

```

I can still do this with an array, but it looks cleaner without it